Development refactor/listeners

KDwevedi opened the pull request.

tushar5526 commented on cogs/listeners/member_events_cog.py:
```diff
@@ -0,0 +1,25 @@
+
+    @commands.Cog.listener("on_member_join")
+    async def on_member_join(self, member: discord.Member):
+        SupabaseInterface().updateContributor(member)
```
Interfaces are generally meant to be extended and not make instances of. Maybe you can refactor to making a helper class and define static methods on it.

KDwevedi replied:
refactored into helpers/supabaseClient

tushar5526 commented on interfaces/supabase.py:
```diff
@@ -29,19 +31,33 @@ def update(self, update, query_key, query_value):
     def insert(self, data):
         data = self.client.table(self.table).insert(data).execute()
```
Possible case of `self.table` being equal to `''`

KDwevedi replied:
addressed in supabase refactor, class no longer specifies table on initialisation

tushar5526 commented on helpers.py:
```diff
@@ -0,0 +1,18 @@
+import discord
```
Import not getting used ?

KDwevedi replied:
made import more specific

tushar5526 commented:
A bunch of observations:

- Generally, python codes use `camel_case` for variable and function declaration. 
- I see type hints in some places. You can use mypy to catch a few more bugs proactively. 
- I can see a lot of unnecessary imports. You can use a combination of `isort` and `flake8` to fix these.
- Also, it would be better integrate `black` for code formatting. 

These linters can be added in as a `pre-commit` and in `GHAs` 

Here is a PR you can refer to for pre-commit setup - https://github.com/Flagsmith/flagsmith-python-client/pull/70
Also attaching a link for Abstrac Base Classes (ABCs) in Python - https://realpython.com/inheritance-composition-python

tushar5526 reviewed (changes requested):
Interfaces are generally meant to be extended and not make instances of. Maybe you can refactor to making a helper class and define static methods on it.

KDwevedi commented:
Tasks:
- [ ] Refactor Supabase interface into a helper class
- [ ] linters, camelCase
- [ ] fixing imports

KDwevedi commented:
Hey @tushar5526 made some changes
- set up flake8, black, isort
- cleaned imports
- refactored supabase class as supabaseClient

Could you take another look?

tushar5526 commented:
Looks fine to me, a bit more nitpicking but I see `Supabase().function_call()` being used at multiple places. Basically you are trying to namespace all the Supabase helper functions at one place, therefore defining those methods as static methods is much more cleaner.

Also marking the `Singleton` pattern if you want a global instance of Supabase being used. I am approving the PR - you can do these changes as per the priority of this ticket. Thanks!

KDwevedi commented:
Appreciate the nitpicking!
